tests/app/celery/test_process_delivery_status_result_tasks.py

```diff
@@ -17,196 +26,358 @@ def sample_delivery_status_result_message():
+                   "3RhdHVzPWRlbGl2ZXJlZCZUbz0lMkIxMTExMTExMTExMSZNZXNzYWdlU2lkPVNNeXl5JkFjY291bnRTaWQ9QUN6enom"
+                   "RnJvbT0lMkIxMjIyMzMzNDQ0NCZBcGlWZXJzaW9uPTIwMTAtMDQtMDE=",
+        'reference': 'SMyyy',
+        'record_status': 'delivered'
+    }
+
+
+@pytest.fixture()
+def sample_sqs_message():
+    return {
+        "body": "UmF3RGxyRG9uZURhdGU9MjMwMzIyMjMzOCZTbXNTaWQ9U014eHgmU21zU3RhdHVzPWRlbGl2ZXJlZCZNZXNzYWdlU3RhdHV"
+                "zPWRlbGl2ZXJlZCZUbz0lMkIxMTExMTExMTExMSZNZXNzYWdlU2lkPVNNeXl5JkFjY291bnRTaWQ9QUN6enomRnJvbT0lMkIx"
+                "MjIyMzMzNDQ0NCZBcGlWZXJzaW9uPTIwMTAtMDQtMDE="
+    }
+
+
+@pytest.fixture()
+def sample_sqs_message_with_twilio_provider():
+    return {
+        "body": "UmF3RGxyRG9uZURhdGU9MjMwMzIyMjMzOCZTbXNTaWQ9U014eHgmU21zU3RhdHVzPWRlbGl2ZXJlZCZNZXNzYWdlU3RhdHV"
+                "zPWRlbGl2ZXJlZCZUbz0lMkIxMTExMTExMTExMSZNZXNzYWdlU2lkPVNNeXl5JkFjY291bnRTaWQ9QUN6enomRnJvbT0lMkIx"
+                "MjIyMzMzNDQ0NCZBcGlWZXJzaW9uPTIwMTAtMDQtMDE=",
+        "provider": "twilio"
+    }
+
+
+@pytest.fixture()
+def sample_sqs_message_without_provider():
+    return {
+        "body": "UmF3RGxyRG9uZURhdGU9MjMwMzIyMjMzOCZTbXNTaWQ9U014eHgmU21zU3RhdHVzPWRlbGl2ZXJlZCZNZXNzYWdlU3RhdHV"
+                "zPWRlbGl2ZXJlZCZUbz0lMkIxMTExMTExMTExMSZNZXNzYWdlU2lkPVNNeXl5JkFjY291bnRTaWQ9QUN6enomRnJvbT0lMkIx"
+                "MjIyMzMzNDQ0NCZBcGlWZXJzaW9uPTIwMTAtMDQtMDE="
+    }
+
+
+# confirmed: all below test case are dependent on valid fixtures
+def test_fixture_has_required_attributes(notify_db_session, sample_delivery_status_result_message):
+    """Test that celery event contains "message" """
+    message = sample_delivery_status_result_message.get("message")
+    provider_name = message.get("provider")
+    body = message.get("body")
+
+    assert message is not None
+    assert provider_name is not None
+    assert body is not None
+
+
+# confirmed: task should retry when incoming message does not contain a message attribute
+def test_celery_event_with_missing_message_attribute(notify_db_session, sample_delivery_status_result_message):
     """Test that celery will retry the task if "message" is missing from the CeleryEvent message"""
 
     # remove message (key) from the sample_delivery_status_result_message
     sample_delivery_status_result_message.pop("message")
 
-    # mock translate_delivery_status() when called within process_delivery_status_result_tasks()
-    mocker.patch("app.clients")
-    mocker.patch("app.clients.sms.SmsClient")
-    mocker.patch(
-        "app.clients.sms.twilio.TwilioSMSClient.translate_delivery_status",
-        return_value=sample_translate_return_value,
-    )
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks.process_delivery_status(event=sample_delivery_status_result_message)
+
+
+# confirmed: task should retry when incoming message does not contain provider attribute
+# todo: make sure this is still true
```

Comment:
Any TODO in the code should include any ticket necessary to fix it in the future, unless you plan on doing this in this PR, but at a later date, in which case you might make a note of it. Also, TODO needs to capitalized, which IDEs usually render in a different color or just list out all the TODOs for you.

Reply:
Also, the confirmed comment should be deleted.